test_client/src/main/scala/io/reactivesocket/tck/Tests.scala

```diff
@@ -1,13 +1,22 @@
 package io.reactivesocket.tck
 
-object Tests {
-  def runTests(cls : Any) : Unit = {
-    
-    val methods = cls.getClass.getDeclaredMethods.reverse
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+
+object Tests extends ClientDSL {
+  def runTests(cls : Any, writer: PrintWriter) : Unit = {
+    this.writer = writer;
```

Comment:
as mentioned above, feels iffy and easy to forget to set the writer, could we change this somehow?